### PR TITLE
Modified asyn output and input eos to allow 3 characters

### PR DESCRIPTION
**Pull request by esmith1729**

### CODE REVIEW

File: asyn/miscellaneous/asynInterposeEos.c [hunk 1]
```diff
@@ -301,6 +301,7 @@ static asynStatus setInputEos(void *ppvt,asynUser *pasynUser,
         epicsSnprintf(pasynUser->errorMessage,pasynUser->errorMessageSize,
                         "%s illegal eoslen %d", peosPvt->portName,eoslen);
         return asynError;
+	case 3: peosPvt->eosIn[2] = eos[2]; /*fall through to case 2 */
```
Comment (FreddieAkeroyd) [inline asyn/miscellaneous/asynInterposeEos.c:304]: You also need to increase the size of `eosIn` and  `eosOut` in `struct eosPvt`

**Comment by FreddieAkeroyd** [comment]
There is a comment in the original code 
```
                    /* Resynchronize the search.  Since the driver
                     * allows a maximum two-character EOS it doesn't
                     * have to worry about cases like:
                     *    End-of-string is "eeef"
                     *    Input stream so far is "eeeeeeeee"
                     */ 
```
As the size is now > 2 please check if anything else needs adjusting and if there are any other implicit assumptions that a maximum of two characters will be found

**Comment by esmith1729** [comment]
ticket: https://github.com/ISISComputingGroup/IBEX/issues/7005

**Comment by FreddieAkeroyd** [comment]
Another approach used

**Pull request closed without merging**